addons/html_editor/static/src/editor/core/dom_plugin.js

```diff
@@ -323,8 +323,14 @@ export class DomPlugin extends Plugin {
     insertSeparator() {
         const selection = this.shared.getEditableSelection();
         const sep = this.document.createElement("hr");
-        const target = selection.commonAncestorContainer;
-        target.parentElement.before(sep);
+        const element = closestElement(
+            selection.startContainer,
+            'P, PRE, H1, H2, H3, H4, H5, H6, BLOCKQUOTE',
+        );
+
+        if (element && ancestors(element).includes(this.editable)) {
```

Comment:
`closestElement` already checks that you're not beyond the editable root. I guess all you need here is to check if `element` is not the editable itself.
`if (element && element !== this.editable)` would do the trick.

Btw calling `ancestors` without the second argument (the editable) returns the ancestors up to the document root, which is rarely what we want...